syllabus/4-Smart_Contracts/6-Wasm_ink!_slides.md

```diff
@@ -1054,235 +1149,1255 @@ Notes:
+ - validator sees this opportunity and instead proposes his own block, but leaves some of the opportunities out of it.
+ - now other validators are incentivized to take these MEV opportunities and start building on his proposed block, thus causing a re-org.
+
+---
+
+## Consensus instability (longest chain)
+
+- If block rewards are smaller than the MEV opportunities, rational miners / validators will destabilize consensus by reordering or censoring the transactions.
+
+<img style="margin-top: 10px;margin-bottom: 10px" height="600" width="600" src="./img/ink/consensus_instability1.png" />
+
+credit: [Dan Boneh](https://crypto.stanford.edu/~dabo/)
+
+- What could the searcher / arbitrageurs do to not have their MEV stolen from them?
+
+Notes:
+- Time bandit attack
+- Because of MEV we cannot assume 51% honest majority
+- all actors are rational actors, so they will go for maximal profit
+- imagine you are the searcher and you found this MEV opportunity. What can you do to be sure t have it included?
+
+---
+
+## Private mempools
+
+* MEV actors can contract with miners / validators.
+* They do it for a substantial fee, but at leat this way they still profit.
+* Why does this lead to a horrible outcome for a blockchain?
+
+Notes:
+- they send their tx directly to the miner, skipping the mempool
+ - when its the validator turn to propose a block they include this tx in it (for a fee)
+- can you see any problems with this?
+
+---
+
+## Private mempools
+
+- Pretty soon everbody is sending their txs to a handfull of validator that they trust.
+- It is very hard to become a new trusted validator.
+- Massive centralization.
+
+Notes:
+- centralization of power
+- rich get richer type of a problem
+
+---
+
+
+## Flashbots (MEV-geth)
+
+* flashbots: democratizing MEV Extraction.
+* [MEV-geth](https://github.com/flashbots/mev-geth)
+
+Notes:
+- flashbots is a sealed-bid block space auction mechanism for bidding on transaction order preference.
+
+---
+
+## MEV: the good, the bad and the ugly
+
+* is all MEV inherently bad for the network?
+ - arbitrage
+ - liquidations
+* MEV as a security budget?
+* MEV is unavoidable.
+
+Notes:
+- MEV is wrong and it has to be stopped
+- arbitrage: there are markets that are not equal and arbitrageurs provide a service that keeps the systems healthy
+- liquidations: you want fast loan liquidations (right after a price oracle update that triggers them) and the lenders ot be repayed swiftly.
+- from the economic security perspective: MEV could be an additional source of revenue for the validators.
+ - but it HAS to be distributed equally, else it is an almost existential threat to the network.
+- MEV is a fact of life.
+
+---
+
+## What to do: Proposer-builder separation (PBS)
+
+[MEV-boost](https://boost.flashbots.net/) is an implementation of PBS for post-merge Ethereum
+
+<img style="margin-top: 10px;margin-bottom: 10px" width="800" src="./img/ink/pbs.svg" />
```

Comment:
Image is too big and pushes text out of screen. Maybe replace the width with height and reduce accorginly

Reply:
done